rules/no-single-promise-in-promise-methods.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+const {isMethodCall} = require('./ast/index.js');
+const {
+	getParenthesizedText,
+	isParenthesized,
+	needsSemicolon,
+	shouldAddParenthesesToMemberExpressionObject,
+} = require('./utils/index.js');
+
+const MESSAGE_ID_ERROR = 'no-single-promise-in-promise-methods/error';
+const MESSAGE_ID_SUGGESTION_1 = 'no-single-promise-in-promise-methods/suggestion-1';
+const MESSAGE_ID_SUGGESTION_2 = 'no-single-promise-in-promise-methods/suggestion-2';
+const messages = {
+	[MESSAGE_ID_ERROR]: 'Wrapping a single element array with `Promise.{{method}}()` is unnecessary.',
+	[MESSAGE_ID_SUGGESTION_1]: 'Use the value directly.',
+	[MESSAGE_ID_SUGGESTION_2]: 'Wrap the value with `Promise.resolve()`.',
+};
+const METHODS = ['all', 'any', 'race'];
+
+const isPromiseMethodCallWithSingleElementArray = node =>
+	isMethodCall(node, {
+		object: 'Promise',
+		methods: METHODS,
+		optionalMember: false,
+		optionalCall: false,
+		argumentsLength: 1,
+	})
+	&& node.arguments[0].type === 'ArrayExpression'
+	&& node.arguments[0].elements.length === 1
+	&& node.arguments[0].elements[0] !== null
+	&& node.arguments[0].elements[0].type !== 'SpreadElement';
+
+const wrapText = (sourceCode, node, element, text, prefix, suffix) => {
+	if (prefix || suffix) {
+		return `${prefix}${text}${suffix}`;
+	}
+
+	if (node.parent.type === 'MemberExpression'
+	&& !isParenthesized(element, sourceCode)
+	&& shouldAddParenthesesToMemberExpressionObject(element, sourceCode)) {
+		return `(${text})`;
+	}
```

Comment:
This seems not enough, I haven't test but I guess it will break

```js
await Promise.all([x ** y])
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence#table

Reply:
It's corrected.

Reply:
Not correct, I'll fix.